core/io/file_access_encrypted.cpp

```diff
@@ -214,9 +214,13 @@ uint64_t FileAccessEncrypted::get_buffer(uint8_t *p_dst, uint64_t p_length) cons
 	ERR_FAIL_COND_V_MSG(writing, -1, "File has not been opened in read mode.");
 
 	uint64_t to_copy = MIN(p_length, get_length() - pos);
-	memcpy(p_dst, data.ptr() + pos, to_copy);
-	pos += to_copy;
-
+	if (to_copy) {
+		// Don't call memcpy on a potential nullptr dest
```

Comment:
> // Don't call memcpy on a potential nullptr dest

I guess it's `nullptr` `src` (zero size `data` → null `data.ptr()`), not `dest` (it was checked for on line 213). Calling `memcpy` with zero `count` is not UB, but calling it with `nullptr` `dest` or `src` is.

Reply:
See my next comment. This is the same case.